src/hevm/src/EVM/SymExec.hs

```diff
@@ -413,15 +416,42 @@ simplify e = if (mapExpr go e == e)
       | a == b = (And b c)
       | otherwise = o
 
+    -- Bitwise AND & OR. These MUST preserve bitwise equivalence
+    go o@(And (Lit x) _)
+      | x == 0 = Lit 0
+      | otherwise = o
+    go o@(And _ (Lit x))
+      | x == 0 = Lit 0
+      | otherwise = o
+    go o@(Or (Lit x) b)
+      | x == 0 = b
+      | otherwise = o
+    go o@(Or a (Lit x))
+      | x == 0 = a
+      | otherwise = o
+
+    -- Non-bitwise AND & OR -- no need to preserve bitwise equivalence.
+    go (ITE (And (Lit x) a) t f)
+      | x == 0 = f
+      | otherwise = ITE a t f
+    go (ITE (And a b@(Lit _)) t f) = ITE (And b a) t f
+    go (ITE (Or (Lit x) a) t f)
+      | x == 0 = ITE a t f
+      | otherwise = t
+    go (ITE (Or a b@(Lit _)) t f) = ITE (Or b a) t f
+
     -- we write at least 32, so if x <= 32, it's FALSE
     go o@(EVM.Types.LT (BufLength (WriteWord {})) (Lit x))
       | x <= 32 = Lit 0
       | otherwise = o
     -- we write at least 32, so if x < 32, it's TRUE
-    go o@(EVM.Types.GT (BufLength (WriteWord {})) (Lit x))
+    go o@(EVM.Types.LT (Lit x) (BufLength (WriteWord {})))
```

Comment:
Note: we replace all LT with GT, so this simplification rule had to be adjusted.